src/foamadapter/solver/pimplefoam.py

```diff
@@ -0,0 +1,161 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2023 FoamAdapter authors
+
+from typing import Any
+
+import pybFoam as pyf  # type: ignore[import-not-found]
+from pybFoam import (
+    Info,
+    fvc,
+    fvm,
+    fvScalarMatrix,
+    fvVectorMatrix,
+    surfaceScalarField,
+    volScalarField,
+    volVectorField,
+)
+from pybFoam.turbulence import incompressibleTurbulenceModel, singlePhaseTransportModel  # type: ignore[import-not-found]
+
+
+class CFLNumber:
+    def __init__(self, maxDeltaT: float) -> None:
+        self.criteria: list[Any] = []
+        self.GREAT = 1e30
+        self.SMALL = 1e-15
+        controlDict = pyf.dictionary.read("system/controlDict")
+        self.adjustable = controlDict.get[bool]("adjustTimeStep")
+        self.maxCFL = controlDict.get[float]("maxCo")
+        self.maxDeltaT = maxDeltaT
+
+    def setDelta(
+        self, runTime: pyf.Time, phi: pyf.surfaceScalarField, maxRatio: float = 1.2
+    ) -> None:
+        deltaT = runTime.deltaTValue()
+
+        if not self.criteria:
+            return
+
+        maxCFLNumber, meanCFLNumber = pyf.computeCFLNumber(phi)
+        Info(f"Courant Number mean: {meanCFLNumber}, max: {maxCFLNumber}")
+        ratios = [self.maxCFL / maxCFLNumber]
```

Comment:
I still think this shouldn't be a list. Is there ever a situation you have multiple values?